connection.go

```diff
@@ -132,13 +134,22 @@ func NewConnectionBuilder() *ConnectionBuilder {
 		urlTable: map[string]string{
 			"": DefaultURL,
 		},
-		retryLimit:        retry.DefaultLimit,
-		retryInterval:     retry.DefaultInterval,
-		retryJitter:       retry.DefaultJitter,
-		metricsRegisterer: prometheus.DefaultRegisterer,
+		retryLimit:                          retry.DefaultLimit,
+		retryInterval:                       retry.DefaultInterval,
+		retryJitter:                         retry.DefaultJitter,
+		metricsRegisterer:                   prometheus.DefaultRegisterer,
+		includeDefaultAuthnTransportWrapper: true,
 	}
 }
 
+// NewConnectionBuilder creates a Builder that knows how to create connections
+// without authentication
+func NewUnauthenticatedConnectionBuilder() *ConnectionBuilder {
```

Comment:
An alternative name could be NewNoAuthConnectionBuilder() or NewNoAuthnConnectionBuilder() or NewNoAuthenticationConnectionBuilder().

Let me know if you have any preference instead of the currently set name

Reply:
This works for me

Reply:
I'm fine either way